### PR TITLE
Carets - Jan and Maria - Scrabble

### PR DESCRIPTION
**Pull request by ghost**
# Scrabble
Congratulations! You're submitting your assignment.

## Comprehension Questions

|  Feature 	|   Feedback	|
|---	|---	|
|   Why do you think we chose to make the `score` method in the `Scoring` class a **class method** instead of an **instance method**? | The self.score method was made to be a class method in Scoring rather than an instance method in order to access the methods for scoring, without the need for creating an instance of scoring. If the methods were instance methods, we'd have to instantiate Scoring and make a scoring object whenever we wanted to access the scoring methods. 
Describe an example of a test you wrote for a nominal case.   | In our scoring spec we created test to ensure that a letter was returning correct the value when used in play.
 wrote for an _edge case_. | In our title bag spec, we created a test that would throw an error if anything other than a number was used to draw tiles from the bag.	  	|
|   What was your final test coverage percentage?	|At our last check 100% coverage was showing, but only 5 of the 6 files were appearing in the coverage report. Tried to trouble shoot it with no luck.    	|
|   Is there a specific piece of code you'd like feedback on?	|   Feedback on the Simple Cov files.	|

### CODE REVIEW

**Comment by tildeee** [comment]
# Scrabble
## What We're Looking For

|  Feature  |   Feedback |
|---	|---	|
|  **General**	|   	|
|   Answered comprehension questions  |  x  |
|   Both Teammates contributed to the codebase	|  x  |
|   Regular Commits with meaningful commit messages  |  x  |
|  **Scoring class**	|   	|
|  `score` method	|  x  |
|   Uses appropriate data structure to store the letter score	|  x  |
|   Appropriately handles edge cases	|  x  |
|   Tests `score` edge cases (one-letter word, 7 letter word)	|  x  |
|  `highest_scoring_word` method	|  x  |
|   Appropriately handles edge cases	|  x  |
|   Tests `highest_scoring_word` edge cases (ties, no words, no array)	|  x  |
|  **Player class**	|   	|
|  Uses an array to keep track of words played	|  x  |
|  Uses existing code to create `highest_scoring_word` and `highest_word_score`	|  x  |
|  Returns `true` or `false` from `won?` method |  x  |
|  Tests edge cases on the `play(word)` method for words that cannot be scored |      |
|  **TileBag class**	|   	|
|  Uses a data structure to keep the set of default tiles in the bag	|  x  |
|  `draw_tiles` method uses the parameter to draw a set of random tiles |  x  |
| `tiles_remaining` method returns a count	|  x  |
|  Tests `draw_tiles` edge cases for different parameter inputs	|  tests empty string  |

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      @name = name
+      @plays = []
+      @tiles = []
+    end
+
+    def play(word)
+      if won?
+        return false #Returns false if player has already won
+      else
+        @plays << word #Adds the input word to the plays Array
```
Comment (tildeee) [inline lib/player.rb:16]: doesn't remove the tiles played from the player's tiles (not a requirement)

File: lib/scoring.rb [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        raise ArgumentError.new("Word argument must be a string") if !word.is_a? String
+        word = word.upcase.split("")
+        score = word.map {|letter| LETTER_VALUES[letter.to_sym]}.reduce(:+)
+        word.length == 7 ? score += 50 : score
+      end
+
+      def self.highest_score_from(array_of_words)
+        raise ArgumentError.new("Input must be an Array") if !array_of_words.is_a? Array
+        raise ArgumentError.new("elements inside array must be strings") if !array_of_words.all? { |word| word.is_a? String }
+        array_of_words.reduce do |best, word|
```
Comment (tildeee) [inline lib/scoring.rb:44]: SLICK

File: lib/scoring.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      Y: 4,
+      Z: 10
+    }
+
+
+      def self.score(word)
+        raise ArgumentError.new("Word argument must be a string") if !word.is_a? String
+        word = word.upcase.split("")
+        score = word.map {|letter| LETTER_VALUES[letter.to_sym]}.reduce(:+)
+        word.length == 7 ? score += 50 : score
```
Comment (tildeee) [inline lib/scoring.rb:38]: really cool implementation of this method!

**Comment by tildeee** [comment]
Overall this code has some really cool moments of interesting implementation. Well done
